File: Leetcode practice/80. Remove Duplicates from Sorted Array II.cpp
#include <iostream>
#include <vector>
using namespace std;


class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        //olee12
        if(nums.size()==0) return 0;
        int cur = nums[0];
        int cnt = 1;
        int indx = 1;


        for(int i = 1;i<nums.size();i++){
            if(cur == nums[i] && cnt <2){
                nums[indx++] = nums[i];
                cnt++;
            }else if(cur == nums[i]){
                cnt++;
            }else{
                cur = nums[i];
                nums[indx++] = nums[i];
                cnt = 1; 
            }
        }
        return indx;
    }
};



int main(){

    return 0;
}

File: go-leet-code/problem/80. Remove Duplicates from Sorted Array II.go
package problem

func removeDuplicates(nums []int) int {
	index, i, next := 0, 0, 0

	for i < len(nums) {
		next, nums[index] = i, nums[i]
		index++
		for next < len(nums) && nums[i] == nums[next] {
			next++
		}
		if next-i > 2 {
			nums[index] = nums[i]
			index++
			i = next
		} else {
			i++
		}
	}
	return index
}
